cpp/src/DataStorm/NodeSessionManager.cpp

```diff
@@ -17,33 +17,35 @@ using namespace DataStormContract;
 
 namespace
 {
-    class SessionForwarderI : public Ice::Blobject
+    class SessionForwarder : public Ice::Blobject
     {
     public:
-        SessionForwarderI(shared_ptr<NodeSessionManager> nodeSessionManager)
+        SessionForwarder(shared_ptr<NodeSessionManager> nodeSessionManager)
             : _nodeSessionManager(std::move(nodeSessionManager))
         {
         }
 
-        virtual bool ice_invoke(Ice::ByteSeq inEncaps, Ice::ByteSeq&, const Ice::Current& curr)
+        bool ice_invoke(Ice::ByteSeq inParams, Ice::ByteSeq&, const Ice::Current& current) final
         {
-            auto pos = curr.id.name.find('-');
-            if (pos != string::npos && pos < curr.id.name.length())
+            auto pos = current.id.name.find('-');
+            if (pos != string::npos && pos < current.id.name.length())
```

Comment:
I don't like very much this mapping between target Id and forwarder Id, I guess the advantage is we don't need a servant per forwarder...